core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -1,78 +1,330 @@
+      }
+    }
+    if (!metadata.getSecondaryIndexNames().isEmpty()) {
+      for (String secondaryIndex : metadata.getSecondaryIndexNames()) {
+        makeAttribute(secondaryIndex, metadata, columnsToAttributeDefinitions);
+      }
+    }
+    requestBuilder.attributeDefinitions(columnsToAttributeDefinitions);
+
+    // build keys
+    buildPrimaryKey(requestBuilder, metadata);
+
+    // build local indexes that corresponding to clustering keys
+    buildLocalIndexes(namespace, table, requestBuilder, metadata);
+
+    // build secondary indexes
+    long ru = DEFAULT_RU;
+    if (options.containsKey(REQUEST_UNIT)) {
+      ru = Long.parseLong(options.get(REQUEST_UNIT));
+    }
+    buildGlobalIndexes(namespace, table, requestBuilder, metadata, ru);
+
+    // ru
+    requestBuilder.provisionedThroughput(
+        ProvisionedThroughput.builder().readCapacityUnits(ru).writeCapacityUnits(ru).build());
+
+    // table name
+    requestBuilder.tableName(getFullTableName(namespacePrefix, namespace, table));
+
+    // create table
+    try {
+      client.createTable(requestBuilder.build());
+      metadataManager.addTableMetadata(namespace, table, metadata);
+    } catch (Exception e) {
+      throw new ExecutionException("creating table failed", e);
+    }
+
+    while (true) {
+      try {
+        Uninterruptibles.sleepUninterruptibly(CREATING_WAITING_TIME, TimeUnit.MILLISECONDS);
+        DescribeTableRequest describeTableRequest =
+            DescribeTableRequest.builder()
+                .tableName(getFullTableName(namespacePrefix, namespace, table))
+                .build();
+        DescribeTableResponse describeTableResponse = client.describeTable(describeTableRequest);
+        if (describeTableResponse.table().tableStatus() == TableStatus.ACTIVE) {
+          break;
+        }
+      } catch (DynamoDbException e) {
+        throw new ExecutionException("getting table description failed", e);
+      }
+    }
+
+    // scaling control
+    boolean noScaling = DEFAULT_NO_SCALING;
+    if (options.containsKey(NO_SCALING)) {
+      if (Boolean.parseBoolean(options.get(NO_SCALING))) {
+        noScaling = true;
+      }
+    }
+    if (!noScaling) {
+      enableAutoScaling(namespace, table, metadata.getSecondaryIndexNames(), ru);
+    }
+
+    // backup control
+    boolean noBackup = DEFAULT_NO_BACKUP;
+    if (options.containsKey(NO_BACKUP)) {
+      if (Boolean.parseBoolean(options.get(NO_BACKUP))) {
+        noBackup = true;
+      }
+    }
+    if (!noBackup) {
+      enableContinuousBackup(namespace, table);
+    }
   }
 
   @Override
   public void dropTable(String namespace, String table) throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+    DeleteTableRequest request =
+        DeleteTableRequest.builder()
+            .tableName(getFullTableName(namespacePrefix, namespace, table))
+            .build();
+    try {
+      client.deleteTable(request);
```

Comment:
Looks like the original schema loader disables auto scaling before dropping the table?
https://github.com/scalar-labs/scalardb/blob/98fc5103dd10d2a2fb60ea18f888da443d362668/tools/scalar-schema/src/scalar_schema/dynamo.clj#L228-L229

Reply:
Thanks for pointing it out. I update the code to support this [4af745c](https://github.com/scalar-labs/scalardb/pull/301/commits/4af745cac2700265b8116b2a25ce0a98dfdbea9e)
Since it's an option that comes from CLI, and we don't have optional params for dropping table so for now dropping table will try to disable auto-scaling first even auto-scaling not existed already and drop table later.

Reply:
Thank you for handle this!

> Since it's an option that comes from CLI, and we don't have optional params for dropping table so for now dropping table will try to disable auto-scaling first even auto-scaling not existed already and drop table later.

So, when dropping a table created with `no-scaling` option, it looks like it shows a warning message, correct? If it's correct, I don't think we should show the message because dropping a table created with `no-scaling` option is a usual way, and the warning message is confusing.

Reply:
The warning message is only shown when the exception is not `ObjectNotFoundException`, so in case the table is created with `no-scaling` option then the dropping table will not show any warning message.